pkg/cosign/tsalog.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cosign
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"fmt"
+	"os"
+
+	"github.com/sigstore/cosign/v2/pkg/cosign/env"
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
+	"github.com/sigstore/sigstore/pkg/tuf"
+)
+
+const (
+	tsaLeafCertStr                = `tsa_leaf.crt.pem`
+	tsaRootCertStr                = `tsa_root.crt.pem`
+	tsaIntermediateCertStrPattern = `tsa_intermediate_%d.crt.pem`
+)
+
+type TSACertificates struct {
+	LeafCert          *x509.Certificate
+	IntermediateCerts []*x509.Certificate
+	RootCert          []*x509.Certificate
+}
+
+type GetTargetStub func(name string) ([]byte, error)
+
+func GetTufTargets(name string) ([]byte, error) {
+	tufClient, _ := tuf.NewFromEnv(context.Background())
+	return tufClient.GetTarget(name)
+}
+
+// GetTSACerts retrieves trusted TSA certificates from the embedded or cached
+// TUF root. If expired, makes a network call to retrieve the updated targets.
+// By default, the certificates come from TUF, but you can override this for test
+// purposes by using an env variable `SIGSTORE_TSA_CERTIFICATE_FILE` or a file path
+// specified in `TSACertChainPath`. If using an alternate, the file should be in PEM format.
+func GetTSACerts(_ context.Context, certChainPath string, fn GetTargetStub) (*TSACertificates, error) {
```

Comment:
should this context be plumbed through to `GetTufTargets` where `tuf.NewFromEnv` is called?